### DISCUSSION TITLE
how to set tags?

### DISCUSSION THREAD

**Discussion by zaccur** [squidfunk/mkdocs-material#6286] [Q&A] [answered]
![image](https://github.com/squidfunk/mkdocs-material/assets/2202422/b5c28414-5ecd-4ceb-b6af-d91512a71c6d)

Please Help~
I can't find how to set this in docs

**Comment by alexvoss** [comment]
These tags are a customization of the Material documentation. They are not a feature of Material itself. If you are a sponsor then you can look at the [implementation here](https://github.com/squidfunk/mkdocs-material-insiders/blob/master/src/overrides/hooks/shortcodes.py). Essentially, you would configure an `on_page_markdown` hook and replace some placeholder string with appropriate HTML, such as a span, which you can then embed an icon and text in. Then style with CSS. 

There was a similar discussion the other day but I cannot seem to find it just now, will add if I do.

**Reply by squidfunk** [reply, marked as answer]
In fact, the implementation is [part of the community repository](https://github.com/squidfunk/mkdocs-material/blob/master/src/overrides/hooks/shortcodes.py), so you don't need to be a sponsor to take and adapt our custom hook ☺️ Other discussion was at #6020. We might generalize this in the future to ship it as a native component, but we currently have other priorities. Will re-evaluate in the future if more users need it.

**Reply by zaccur** [reply]
thanks

**Reply by pyhsudars** [reply]
Tags as a native component would be good in the community edition

**Reply by squidfunk** [reply]
Thanks for your feedback – we'll keep that in mind ☺️

**Comment by gmkado** [comment]
I was able to get a version of this working using the [badges plugin](https://mkdocs-badges.six-two.dev/badges/).  Combined with the [magic link extension](https://facelessuser.github.io/pymdown-extensions/extensions/magiclink/) I can write something like this:

```md
|:material-tag:|#499|
```

And it renders like this:

![image](https://github.com/squidfunk/mkdocs-material/assets/6592135/d62363f0-bf9b-4bbf-9f98-dab3e3ab602d)

And is clickable and takes me to my pull request matching that number.

**Comment by Aetherinox** [comment]
The badges plugin is nice, however, the ones used in the mkdocs material documentation have a very small footprint.

So after a few hours of messing with the hooks, I was able to implement it.

![IaxHuEl](https://github.com/squidfunk/mkdocs-material/assets/118329232/7c1787b3-8452-4f94-9615-13f5ef61c1e3)


Would be nice to get it as a feature one day, but for now, this works good.

Just a matter of getting the icons right. For icons that small, I tried Fontawesome 6's duo-tone, and they seem too small for that type of font.
